### ISSUE TITLE
[Android] Allow adb kill/start bypassing

### ISSUE THREAD

**Issue by kLabz** [openfl/lime#313]
Hi,

I have some issues with a new linux install where adb start-server needs to be run as root so that other adb commands connect to my devices. I know there is a workaround since I used it in a previous linux install, but I don't remember it and it could be useful not to restart adb everytime anyway.

Thing is, lime now automatically kills/starts adb, so it can't connect to my devices afterwards. I commented lines 85 & 86 of `AndroidHelper` to make it work again (and recompiled lime tools):
```haxe
// need to wake up ADB, this shouldn't be necessary :(
ProcessHelper.runCommand (adbPath, adbName, [ "kill-server" ]);
ProcessHelper.runCommand (adbPath, adbName, [ "start-server" ]);
```

Could you add an argument to lime test / lime run, something like `-no-adb-restart` which only purpose would be to skip those two lines? Or some other way to bypass those two lines?

Thanks!

**Comment by jgranick** [comment]
ADB has been incredibly stubborn in this piece of code. In order to add the "-d" (downgrade) flag, you need to ensure that the connected device is new enough, because it actually requires the feature to be present on the device's own application manager. Pity.

It's quite common if you use common sample names (like DisplayingABitmap) and especially if you build from multiple systems, that it might refuse to install due to an "older version" being installed. This is due to the build number that is incremented on your system, each time you build.

I've had many instances with ADB hanging perpetually when trying to ask what OS version the device is running, mostly on OS X. I have tracked this to a different between `Sys.command` and `sys.io.Process` in Haxe -- when I use a system command, it works as expected, but ADB gets bizarre when it is run as a child process, and perpetually may hang, freezing the tools. I've been hunting a good solution for this for far too long :frowning: 

A day or two ago, I had the idea to use `Sys.command`, but to pipe the output to a file. This allows me to silently try to retrieve the device version, without using `sys.io.Process` to do it. This seems to resolve the weird scenario where calling `adb` would not sort out the ADB server properly, so the kill/start commands (that I never intended to have to begin with) are unneeded.

I also found a way to set a timeout on this command on OS X, since checking the device version is not so important that everything should hang on this. I think I have some combination of an old device and VMWare that get upset about communicating.

Anyway, long story short, I think I have things working better for OS X now, hopefully this is finally dealt with in a reasonable manner, but I'm always interested in knowing anything that can be done in the tools to help make things work out-of-the-box, so please keep it/me in mind as you continue to work through things on your system and device :smile:

Thanks!

**Comment by kLabz** [comment]
Thanks :) That was a quick one!

My environment seems to be pretty much operational now, don't see much issues with lime tools.

I may come back for lime next/legacy issues later, though. I'm having a hard time with it on native targets (messed-up rendering with next (linux), font rendering issues with legacy (android/linux), android soft keyboard issues, etc.) =/
